Rubberduck.VBEEditor/Application/HostApplicationBase.cs

```diff
@@ -1,54 +1,60 @@
-﻿using System;
-using System.Linq;
-using System.Runtime.InteropServices;
-using NLog;
+﻿using System.Runtime.InteropServices;
 using Rubberduck.VBEditor.SafeComWrappers;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
 
 namespace Rubberduck.VBEditor.Application
 {
     [ComVisible(false)]
-    public abstract class HostApplicationBase<TApplication> : IHostApplication
+    public abstract class HostApplicationBase<TApplication> : SafeComWrapper<TApplication>, IHostApplication
         where TApplication : class
     {
-        private static readonly Logger _logger = LogManager.GetCurrentClassLogger();
-
-        protected readonly TApplication Application;
         protected HostApplicationBase(string applicationName)
+        :base(ApplicationFomComReflection(applicationName))
         {
             ApplicationName = applicationName;
+        }
+
+        protected HostApplicationBase(IVBE vbe, string applicationName)
+        :base(ApplicationFomVbe(vbe, applicationName))
+        {
+            ApplicationName = applicationName;
+        }
 
+        private static TApplication ApplicationFomComReflection(string applicationName)
```

Comment:
I believe it's misspelled, missing `r` for the `From`  --- `ApplicationFromComReflection`